submit-api/src/submit_api/services/submission/submission_creator_factory.py

```diff
@@ -119,6 +123,112 @@ def replace(self, submission_id, request_data):
+        session.add(submission)
+
+        return new_submission
+
+    @staticmethod
+    def _restore_previous_active_submission(session, moved_submission):
+        """Restore the most recent previous version if none are currently active."""
+        # Check if there is any other active submission for the same root_submission_id
+        active_exists = session.query(SubmissionModel).filter(
+            SubmissionModel.root_submission_id == moved_submission.root_submission_id,
+            SubmissionModel.id != moved_submission.id,
+            SubmissionModel.deleted.is_(False),
+            SubmissionModel.active.is_(True)
+        ).first()
+
+        if active_exists:
+            # No need to restore — an active submission already exists
+            return
+
+        # Find the latest previous non-deleted submission
+        previous_version = (
+            session.query(SubmissionModel)
+            .filter(
+                SubmissionModel.item_id == moved_submission.item_id,
+                SubmissionModel.root_submission_id == moved_submission.root_submission_id,
+                SubmissionModel.id != moved_submission.id,
+                SubmissionModel.deleted.is_(False),
+                SubmissionModel.active.is_(False),
+                SubmissionModel.minor_version == moved_submission.minor_version - 1
+            )
+            .first()
+        )
+
+        if not previous_version.active:
+            previous_version.active = True
+            session.add(previous_version)
+
+        return
+
+    def _move_to_folder(self, session, submission, request_data):
+        """Move document to a specific folder."""
+        # Check if there is any other active submission for the same root_submission_id
+        if status := submission.status not in [SubmissionStatus.SUBMITTED,
+                                               SubmissionStatus.REJECTED,
+                                               SubmissionStatus.PENDING, SubmissionStatus.PENDING_REPLACEMENT]:
+            raise BadRequestError(f"Cannot replace a document with status {status}.")
+
+        self._fill_missing_name(request_data, submission)
+        submitted_document = self._create_submitted_document(session, request_data)
+
+        new_submission = self._create_submission(
+            session=session,
+            item_id=request_data.get('item_id'),
+            submitted_document_id=submitted_document.id
+        )
+        submission.active = False
+        submission.deleted = True
+
+        session.add(submission)
+
+        return new_submission
+
+    def move(self, submission_id, request_data):
+        """Move a document submission."""
+        with session_scope() as session:
+            submission: SubmissionModel = SubmissionModel.find_by_id(submission_id)
+
+            if request_data.get("target_submission_id"):
+                moved_submission = self._create_next_version_of_target(session, submission, request_data)
```

Comment:
I think we might also nee the "self._restore_previous_active_submission(session, submission)" in the first one. What if we split from a stack and merge onto another

Reply:
Thanks, added it.